File: u4-ruby-on-rails/d19-constructor/class-notes/person.rb
class Person

  # - With an options Hash
  # def initialize(options={})
  #   @first_name = options[:first_name]
  #   @last_name = options[:last_name]
  #   @age = age
  # end

  # - Using Keyword Arguments
  # def initialize(first_name: nil, last_name: nil, age: nil)
  #   @first_name = first_name
  #   @last_name = last_name
  #   @age = age
  # end

  def initialize(first_name, last_name, age)
    @first_name = first_name
    @last_name = last_name
    @age = age
  end

  def full_name
    [@first_name, @last_name].join(' ')
  end

  def greeting
    "Hello #{full_name}"
  end
end

File: u4-ruby-on-rails/d19-constructor/class-notes/person.js
// Person class with an options Hash
/*
var Person = function(options) {
  if (options === undefined) { options = {} }
  this.firstName = options['firstName'];
  this.lastName = options['lastName'];
  this.age = options['age'];
}
*/

var Person = function(firstName, lastName, age) {
  this.firstName = firstName;
  this.lastName = lastName;
  this.age = age;
}

Person.prototype.fullName = function() {
  return this.firstName + ' ' + this.lastName;
}

Person.prototype.greeting = function() {
  return "Hello "+this.fullName();
}
